src/full/Agda/TypeChecking/Substitute/Class.hs

```diff
@@ -109,6 +111,13 @@ instance Subst QName where
   type SubstArg QName = Term
   applySubst _ q = q
 
+newtype NoSubst t a = NoSubst { unNoSubst :: a }
+  deriving (Generic, NFData, Functor)
```

Comment:
Please equip this new phantom type with a statement of purpose, i.e. what are its intended uses.

Reply:
Thanks!